src/spikeinterface/core/basesorting.py

```diff
@@ -347,7 +412,7 @@ def get_all_spike_trains(self, outputs="unit_id"):
             spikes.append((spike_times, spike_labels))
         return spikes
 
-    def to_spike_vector(self, extremum_channel_inds=None):
+    def to_spike_vector(self, concatenated=True, extremum_channel_inds=None, use_cache=True):
```

Comment:
concatenated=True/False is truely need. because in many place we want to loop over segments without to cut at the limit.

use_cache is also important in case you want to use it once and do not want to keep it in memory.

extremum_channel is need for spike location computation. This could be removed here and add more complexity on compute_spike_olocation side. Not sure if this is relevant or not.

Reply:
maybe move the new args after `extremum_channel_inds=None`?